framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -896,31 +910,63 @@ protected String dotOutputFileName(UnderlyingAST ast) {
         return null;
     }
 
+    /**
+     * Returns the type of the left-hand side of an assignment without
+     * applying local variable defaults to type variables.
+     *
+     * The type variables that are types of local variables are defaulted to
```

Comment:
>Is the problem that for reference types we don't know what to use instead?  

Yes.

>Couldn't we just use the default that we would use for parameters/returns/fields then?

Something like this would probably work, but it would probably have to be type system dependent.

I've added a comment explaining this:  3c050a6.